supervisor/loggers.py

```diff
@@ -138,11 +139,29 @@ class FileHandler(Handler):
     """File handler which supports reopening of logs.
     """
 
-    def __init__(self, filename, mode="a"):
+    def __init__(self, filename, mode=None):
         Handler.__init__(self)
-        self.stream = open(filename, mode)
+        open_mode = mode or 'a'
+
+        try:
+            self.stream = open(filename, open_mode)
+        except OSError as e:
+            retried = False
+            if not mode:
```

Comment:
I took me a couple passes through but I think what this dual `mode` and `open_mode` is intended to do is "If nothing was given for mode, try 'a' and then try 'w' if the Python 3 bug occurs.  But if a specific mode was supplied, try only that mode and don't work around the Python 3 bug".

Is there a case where having the workaround for the Python 3 implicit SEEK would be undesirable?  If `'w'` and `'a'` really do have the same semantics on these devices, we can simplify this back to `mode` and remove the special `if not mode` case.  I find the dual `mode`/`open_mode` pretty confusing.  If we do need to keep it, we should probably add some comments explaining why we need the separate behaviors.